sponsorship_compassion/model/contracts.py

```diff
@@ -581,83 +540,76 @@ def contract_waiting(self, cr, uid, ids, context=None):
     ##########################################################################
-    def _on_sponsorship_finished(self, cr, uid, sponsorship_ids,
-                                 context=None):
+
+    @api.multi
+    def _on_sponsorship_finished(self, sponsorships):
         """ Called when a sponsorship is terminated or cancelled:
         Remove sponsor from the child, terminate related gift
         contracts, and remove sponsor category if sponsor has no other
         active sponsorships.
         """
-        ctx = {'lang': 'en_US'}
-        category_obj = self.pool.get('res.partner.category')
-        sponsor_cat_id = category_obj.search(
-            cr, uid, [('name', '=', 'Sponsor')], context=ctx)[0]
+        category_obj = self.env['res.partner.category'].with_context(
+            lang='en_US')
+
+        sponsor_cat_id = category_obj.search([('name', '=', 'Sponsor')])[0].id
         old_sponsor_cat_id = category_obj.search(
-            cr, uid, [('name', '=', 'Old Sponsor')],
-            context=ctx)[0]
+            [('name', '=', 'Old Sponsor')])[0].id
         wf_service = netsvc.LocalService('workflow')
-        con_line_obj = self.pool.get('recurring.contract.line')
 
-        for sponsorship in self.browse(cr, uid, sponsorship_ids, context):
+        for sponsorship in sponsorships:
             sponsorship.child_id.write({'sponsor_id': False})
 
-            con_ids = self.search(cr, uid, [
+            contract_count = self.env['recurring.contract'].search_count([
                 ('partner_id', '=', sponsorship.partner_id.id),
                 ('state', '=', 'active'),
-                ('type', 'like', 'S')], context=context)
-            if not con_ids:
+                ('type', 'like', 'S')])
+            if not contract_count:
                 # Replace sponsor category by old sponsor category
                 sponsorship.partner_id.write({
                     'category_id': [(3, sponsor_cat_id),
                                     (4, old_sponsor_cat_id)]})
 
-            gift_con_ids = con_line_obj.search(cr, uid, [
-                ('sponsorship_id', '=', sponsorship.id)], context=context)
-            for line in con_line_obj.browse(cr, uid, gift_con_ids, context):
+            gift_contract_lines = self.env['recurring.contract.line'].search([
+                ('sponsorship_id', '=', sponsorship.id)])
+            for line in gift_contract_lines:
                 contract = line.contract_id
                 if len(contract.contract_line_ids) > 1:
                     line.unlink()
                 else:
                     wf_service.trg_validate(
-                        uid, self._name, contract.id,
-                        'contract_terminated', cr)
+                        self.env.user.id, self._name, contract.id,
+                        'contract_terminated', self.env.cr)
 
-    def _on_contract_active(self, cr, uid, ids, context=None):
+    @api.multi
+    def _on_contract_active(self):

```

Comment:
This method no longer exist. Please check contract_compassion module (and changes between v7 and v8). The code should be moved in `contract_active` method I guess.